### ISSUE TITLE
NilClass error on CircleCI

### ISSUE THREAD

**Issue by bright-star** [hanneskaeufler/danger-todoist#32]
## Log

```
bundle exec danger

[!] The exception involves the following plugins:
 -  danger-todoist

bundler: failed to load command: danger (/home/ubuntu/repo/vendor/bundle/ruby/2.2.0/bin/danger)
Danger::DSLError: 
[!] Invalid `Dangerfile` file: undefined method `[]' for nil:NilClass
 #  from Dangerfile:29
 #  -------------------------------------------
 #  # Warn if TODOs are present in code
 >  todoist.warn_for_todos
 #  todoist.print_todos_table
 #  -------------------------------------------

  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/git-1.3.0/lib/git/diff.rb:64:in `[]'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/danger-4.0.5/lib/danger/danger_core/plugins/dangerfile_git_plugin.rb:115:in `diff_for_file'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/danger-todoist-1.2.0/lib/todoist/plugin.rb:132:in `block in diffs_of_interest'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/danger-todoist-1.2.0/lib/todoist/plugin.rb:132:in `map'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/danger-todoist-1.2.0/lib/todoist/plugin.rb:132:in `diffs_of_interest'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/danger-todoist-1.2.0/lib/todoist/plugin.rb:112:in `find_todos'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/danger-todoist-1.2.0/lib/todoist/plugin.rb:103:in `call_method_for_todos'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/danger-todoist-1.2.0/lib/todoist/plugin.rb:55:in `warn_for_todos'
  Dangerfile:29:in `block in parse'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/danger-4.0.5/lib/danger/danger_core/dangerfile.rb:199:in `eval'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/danger-4.0.5/lib/danger/danger_core/dangerfile.rb:199:in `block in parse'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/danger-4.0.5/lib/danger/danger_core/dangerfile.rb:195:in `instance_eval'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/danger-4.0.5/lib/danger/danger_core/dangerfile.rb:195:in `parse'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/danger-4.0.5/lib/danger/danger_core/dangerfile.rb:272:in `run'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/danger-4.0.5/lib/danger/danger_core/executor.rb:27:in `run'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/danger-4.0.5/lib/danger/commands/runner.rb:66:in `run'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/claide-1.0.1/lib/claide/command.rb:334:in `run'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/gems/danger-4.0.5/bin/danger:5:in `<top (required)>'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/bin/danger:23:in `load'
  /home/ubuntu/repo/vendor/bundle/ruby/2.2.0/bin/danger:23:in `<top (required)>'

bundle exec danger returned exit code 1
```

## Dangerfile

```ruby
# Sometimes it's a README fix, or something like that - which isn't relevant for
# including in a project's CHANGELOG for example
# declared_trivial = github.pr_title.include? "#trivial"

# Make it more obvious that a PR is a work in progress and shouldn't be merged yet
warn("PR is classed as Work in Progress") if github.pr_title.include? "[WIP]"

# Warn when there is a big PR
warn("Big PR") if git.lines_of_code > 500

# Don't let testing shortcuts get into master by accident
# fail("fdescribe left in tests") if `grep -r fdescribe specs/ `.length > 1
# fail("fit left in tests") if `grep -r fit specs/ `.length > 1

# Ensure a PR has someone assigned to it
warn "This PR does not have any assignees yet." unless github.pr_json["assignee"]

# PRs should reference an issue
unless (github.pr_body.include? "Closes #" or github.pr_body.include? "Connects #")
  fail "This PR doesn't close or connect to an issue."
end

# PRs should almost always merge to dev.
unless github.branch_for_base == "dev"
  warn "This PR will not be merged into dev."
end

# Warn if TODOs are present in code
todoist.warn_for_todos
todoist.print_todos_table

# Lint commit message
commit_lint.check

# Mention potential reviewers
mention.run
```

## Gemfile

```ruby
# frozen_string_literal: true
source "https://rubygems.org"

gem 'danger'
gem 'danger-todoist' # Checks for TODO etc in commits
gem 'danger-commit_lint' # Lints commit messages
gem 'danger-mention' # Mention potential reviewers
```

**Comment by hanneskaeufler** [comment]
Thanks for reporting this, I sure hoped I git that one (dupe #22) in #24 already, seems theres more to it. I just published v1.2.1, would be awesome if you could upgrade.

**Comment by bright-star** [comment]
That fixed it. Thanks!
